nomad/variables_endpoint.go

```diff
@@ -579,3 +623,62 @@ func (sv *Variables) groupForAlloc(claims *structs.IdentityClaims) (string, erro
 	}
 	return alloc.TaskGroup, nil
 }
+
+// RenewLock is used to apply a SV renew lock operation on a variable to maintain the lease.
+func (sv *Variables) RenewLock(args *structs.VariablesRenewLockRequest, reply *structs.VariablesRenewLockResponse) error {
+	authErr := sv.srv.Authenticate(sv.ctx, args)
+	if authErr != nil {
+		return structs.ErrPermissionDenied
+	}
```

Comment:
The auth err should be handled after forwarding and RPC rate metrics (compare to the `Apply` RPC above). I know that seems a little weird, but this is how we ensure we can get rate metrics even for invalid authentication or on stale followers.

Reply:
Let me look at this better, not very familiar :)

Reply:
It is a weird flow how it is forwarded before checking the error, do you know why it was built like that?